pkg/volume/gce_pd/attacher.go

```diff
@@ -165,6 +165,9 @@ func (attacher *gcePersistentDiskAttacher) MountDevice(spec *volume.Spec, device
 	options := []string{}
 	if readOnly {
 		options = append(options, "ro")
+	} else {
+		// as per https://cloud.google.com/compute/docs/disks/add-persistent-disk#formatting
+		options = append(options, "discard")
```

Comment:
Does this mean we should not mount with trim if we did not mkfs with trim?
 as things stand today, there is no way to pass custom flags to mkfs.  But
it looks like GCE suggests setting it regardless of SSD-ness (since PD is
...unique)

On Wed, Jul 6, 2016 at 11:45 PM, Michael Rubin <notifications@github.com>
wrote:

> In pkg/volume/gce_pd/attacher.go
> <https://github.com/kubernetes/kubernetes/pull/28448#discussion_r69857853>
> :
>
> > @@ -165,6 +165,9 @@ func (attacher *gcePersistentDiskAttacher) MountDevice(spec *volume.Spec, device
> >  	options := []string{}
> >  	if readOnly {
> >  		options = append(options, "ro")
> > +	} else {
> > +		// as per https://cloud.google.com/compute/docs/disks/add-persistent-disk#formatting
> > +		options = append(options, "discard")
>
> So you just made me look at the kernel source code again.
>
> We should not run mkfs if we don't know that the device supports TRIM.
>
> You can see in fs/ext4/mballoc.c that if DISCARD is set but not supported
> in the device the kernel will emit a warning to dmesg, even just for mount.
> I think if we cannot test for SSD or discard supported device we should not
> set this.
> There is no way to set mkfs flags per device?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/28448/files/8efefab9a3fdc4906ca53d4a23d22c3846d0610b#r69857853>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AFVgVICvE6nVIqxXld383IVXtADLn7maks5qTKCYgaJpZM4JECrD>
> .
>